backend/check_stac_metadata/utils.py

```diff
@@ -74,16 +74,21 @@ def run(self, metadata_url: str) -> None:
                 ValidationResult.FAILED,
                 details={"message": error_message},
             )
-            raise AssertionError(error_message)
-        self.validate(metadata_url)
+            LOGGER.error(dumps({"success": False, "message": error_message}))
+            return
 
-    def validate(self, url: str) -> None:  # pylint: disable=too-complex
-        self.traversed_urls.append(url)
         try:
-            object_json = self.get_object(url)
-        except JSONDecodeError:
+            self.validate(metadata_url)
+        except (ValidationError, ClientError, JSONDecodeError) as error:
+            LOGGER.error(dumps({"success": False, "message": str(error)}))
             return
 
+        self.save(hash_key)
```

Comment:
Do we want to inline the `save` method? It's only being called here, as far as I can tell.